substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debuginfo/DebugInfoProvider.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.debuginfo;
+
+import java.util.List;
+
+// class defining interfaces used to allow a native image
```

Comment:
Can you please convert all these C++ style comments to proper javadoc.

Reply:
Yes, of course -- I should have been on the ball from the start.

I have converted almost all of these existing comments to javadoc, updated the content to be more informative and added a lot more missing javadoc comments.

I have (deliberately) not javadoc commented all the final values that clearly identify DWARF constants -- there is the DWARF spec for that.

I have also not (yet) javadoc commented some of the internal details of DwarfSectionImpl and its subclasses, most notably the various putXXX and writeXXX methods that insert simple data records to the ELF sections in a variety of different formats. The higher level methods which invoke them provide comments detailing the data layouts so this omission should not be not critical when it comes to understanding the code.